merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -676,6 +677,45 @@ void extractOutput(final SerializableTopic<T> topic, final Event ev, final SpanI
     }
   }
 
+  /**
+   * Retrieves the activity stack trace from a given span.
+   *
+   * This method analyzes the provided span within the event graph to identify a list of parent activities leading up to it.
+   *
+   * @param engine the simulation engine
+   * @param activityTopic the activity topic
+   * @param serializableTopics a collection of serializable topics
+   * @param spanId the ID of the span to analyze
+   * @return a list of serialized activities representing the stack trace
+   */
+  public List<SerializedActivity> findActivityTraceFromSpan(
```

Comment:
The code for this and `getDirectiveIdFromSpan` overlap quite a bit. Is it possible to collapse them?